src/ILCompiler.CppCodeGen/src/CppCodeGen/ILToCppImporter.cs

```diff
@@ -2633,6 +2633,13 @@ private void ImportLoadField(int token, bool isStatic)
             PushTemp(kind, fieldType);
             AppendCastIfNecessary(kind, fieldType);
 
+            if (fieldType.IsValueType)
+            {
+                Append("*(");
```

Comment:
Would it make sense to emit these extra casts for shared generic code only?

Reply:
yes, these extra casts are needed for shared code only